File: bootpay/src/main/java/kr/co/bootpay/model/RemoteOrderForm.java
package kr.co.bootpay.model;

import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.List;

public class RemoteOrderForm {
    private String m_id = "";
    private String pg = "";
    private List<String> fm = new ArrayList<>();
    private Double tfp = 0.0;
    private String n = "";
    private String cn = "";
    private Double ip = 0.0;
    private Double dp = 0.0;
    private Double dap = 0.0;

    private boolean is_r_n = false;
    private boolean is_r_p = false;
    private boolean is_addr = false;
    private boolean is_da = false;
    private boolean is_memo = false;
    private String desc_html = "";
    private Double dap_jj = 0.0;
    private Double dap_njj = 0.0;
    private String o_key = "";

    public RemoteOrderForm setTaxFreePrice(Double value) {
        this.tfp = value;
        return this;
    }

    public Double getTaxFreePrice() {
        return this.tfp;
    }

    public RemoteOrderForm setPG(String value) {
        this.pg = value;
        return this;
    }

    public String getPG() {
        return this.pg;
    }

    public RemoteOrderForm setMethods(List<String> value) {
        this.fm = value;
        return this;
    }

    public List<String> getMethods() {
        return this.fm;
    }

    public RemoteOrderForm setName(String value) {
        this.n = value;
        return this;
    }

    public String getName() {
        return this.n;
    }

    public RemoteOrderForm setCompanyName(String value) {
        this.cn = value;
        return this;
    }

    public String getCompanyName() {
        return this.cn;
    }

    public RemoteOrderForm setItemPrice(int value) {
        return setItemPrice(new Double(value));
    }

    public RemoteOrderForm setItemPrice(Double value) {
        this.ip = value;
        return this;
    }

    public Double getItemPrice() {
        return this.ip;
    }

    public RemoteOrderForm setDisplayPrice(Double value) {
        this.dp = value;
        return this;
    }

    public Double getDisplayPrice() {
        return this.dp;
    }


    public RemoteOrderForm setDeliveryAreaPrice(int value) {
        return setDeliveryAreaPrice(new Double(value));
    }

    public RemoteOrderForm setDeliveryAreaPrice(Double value) {
        this.dap = value;
        return this;
    }

    public Double getDeliveryAreaPrice() {
        return this.dap;
    }

    public RemoteOrderForm setIsReceiverName(Boolean value) {
        this.is_r_n = value;
        return this;
    }

    public Boolean getIsReceiverName() {
        return this.is_r_n;
    }

    public RemoteOrderForm setIsReceiverPrice(Boolean value) {
        this.is_r_p = value;
        return this;
    }

    public Boolean getIsReceiverPrice() {
        return this.is_r_p;
    }

    public RemoteOrderForm setIsAddr(Boolean value) {
        this.is_addr = value;
        return this;
    }

    public Boolean getIsAddr() {
        return this.is_addr;
    }

    public RemoteOrderForm setIsDeliveryArea(Boolean value) {
        this.is_da = value;
        return this;
    }

    public Boolean getIsDeliveryArea() {
        return this.is_da;
    }

    public RemoteOrderForm setIsMemo(Boolean value) {
        this.is_memo = value;
        return this;
    }

    public Boolean getIsMemo() {
        return this.is_memo;
    }


    public RemoteOrderForm setDescHtml(String value) {
        this.desc_html = value;
        return this;
    }

    public String getDescHtml() {
        return this.desc_html;
    }

    public RemoteOrderForm setDapJeju(int value) {
        return setDapJeju(new Double(value));
    }

    public RemoteOrderForm setDapJeju(Double value) {
        this.dap_jj = value;
        return this;
    }

    public Double getDapJeju() {
        return this.dap_jj;
    }

    public RemoteOrderForm setDapNonJeju(int value) {
        return setDapNonJeju(new Double(value));
    }

    public RemoteOrderForm setDapNonJeju(Double value) {
        this.dap_njj = value;
        return this;
    }

    public Double getDapNonJeju() {
        return this.dap_njj;
    }

    public RemoteOrderForm setOrderFormKey(String value) {
        this.o_key = value;
        return this;
    }

    public String getOrderFormKey() {
        return this.o_key;
    }

    public final String toGson() {
        if(this == null) return "";
        return new Gson().toJson(this);
    }
}

File: bootpay/src/main/java/kr/co/bootpay/model/RemoteOrderForm.kt
package kr.co.bootpay.model

import com.google.gson.Gson


class RemoteOrderForm {
    // request fields
    var m_id = ""
    var pg = ""
    var fm = mutableListOf<String>() //filtered_methods, 결제수단에 사용할 method - card, bank, vbank, phone
    var tfp = 0.0 //tax_free_price
    var n = "" //상품명, name
    var cn = "" //company_name, 업체명
    var ip = 0.0 // item price - 상품가격
    var dp = 0.0  // display price - 할인된 가격을 표시하기 위함
    var dap = 0.0 // 기본배송비

    var is_r_n = false //구매자명 입력을 허용할지 말지, is_receive_name
    var is_r_p = false //구매자가 가격 입력 허용할지 말지
    var is_addr = false // 주소 받을지 말지
    var is_da = false // 배송비 받을지 말지
    var is_memo = false

    var desc_html = ""
    var dap_jj = 0.0 // 제주지역 도서산간 추가비용
    var dap_njj = 0.0 // 제주 외 지역 도서산간 추가비용
    var o_key = ""

    fun setTaxFreePrice(value: Double): RemoteOrderForm {
        tfp = value
        return this
    }

    fun setPG(value: String): RemoteOrderForm {
        pg = value
        return this
    }

    fun setMethods(value: MutableList<String>): RemoteOrderForm {
        fm = value
        return this
    }

    fun setName(value: String): RemoteOrderForm {
        n = value
        return this
    }

    fun setCompanyName(value: String): RemoteOrderForm {
        cn = value
        return this
    }

    fun setItemPrice(value: Double): RemoteOrderForm {
        ip = value
        return this
    }

    //Deprecated 지금은 사용하지 않음
    fun setDisplayPrice(value: Double): RemoteOrderForm {
        dp = value
        return this
    }

    fun setDeliveryAreaPrice(value: Double): RemoteOrderForm {
        dap = value
        return this
    }

    fun setIsReceiverName(value: Boolean): RemoteOrderForm {
        is_r_n = value
        return this
    }

    fun setIsReceiverPrice(value: Boolean): RemoteOrderForm {
        is_r_p = value
        return this
    }

    fun setIsAddr(value: Boolean): RemoteOrderForm {
        is_addr = value
        return this
    }

    fun setIsDeliveryArea(value: Boolean): RemoteOrderForm {
        is_da = value
        return this
    }

    fun setIsMemo(value: Boolean): RemoteOrderForm {
        is_memo = value
        return this
    }

    fun setDescHtml(value: String): RemoteOrderForm {
        desc_html = value
        return this
    }

    fun setDapJeju(value: Double): RemoteOrderForm {
        dap_jj = value
        return this
    }

    fun setDapNonJeju(value: Double): RemoteOrderForm {
        dap_njj = value
        return this
    }

    fun setObjectKey(value: String): RemoteOrderForm {
        o_key = value
        return this
    }

    fun toGson() = Gson().toJson(this)
}
